tools/autograd/templates/Functions.cpp

```diff
@@ -364,17 +364,20 @@ Tensor unsqueeze_to(const Tensor & self, int64_t dim, IntList sizes) {
   return self;
 }
 
-std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<int64_t> &sizes, int64_t dim) {
+std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, int64_t dim) {
```

Comment:
Hmmmmmmmm. You told me that this wasn't going to be part of the ATen API, which was true, but it's part of the "code in Functions.cpp which morally should be in ATen but isn't because shenanigans".

Last time this happened to me, I think I ended up passing the entire list of tensors to the function in question, because I secretly needed them anyway for something else. That's clearly not appropriate here. I guess this implies we should get something like `std::vector<TensorGeometry>` or something similar working in the codegen.